packages/@o3r/testing/src/core/playwright/component-fixture.ts

```diff
@@ -148,38 +142,29 @@ export class O3rComponentFixture<V extends O3rElement = O3rElement> implements C
     timeout?: number;
   } = {}): Promise<void> {
     const element = await this.queryWithOptions(selector, options.elementConstructor, options);
-    if (!!element && await element.isVisible()) {
+    if (await element.isVisible()) {
       await element.click();
     }
   }
 
   /** @inheritdoc */
-  public async query(selector: string, returnType?: undefined): Promise<O3rElement | undefined>;
-  public async query<T extends O3rElement>(selector: string, returnType: O3rElementConstructor<T>): Promise<T | undefined>;
-  public async query<T extends O3rElement>(selector: string, returnType: O3rElementConstructor<T> | undefined): Promise<T | O3rElement | undefined> {
-    try {
-      const elements = this.rootElement.sourceElement.element.locator(selector);
-      const element = elements.first();
-      const selectedElement: PlaywrightSourceElement = {element: element, page: this.rootElement.sourceElement.page};
-      return Promise.resolve(new (returnType || O3rElement)(selectedElement));
-    } catch (err) {
-      console.warn(`Failed to query ${selector}`, err);
-      return Promise.resolve(undefined);
-    }
+  public async query(selector: string, returnType?: undefined): Promise<O3rElement>;
+  public async query<T extends O3rElement>(selector: string, returnType: O3rElementConstructor<T>): Promise<T>;
+  public async query<T extends O3rElement>(selector: string, returnType: O3rElementConstructor<T> | undefined): Promise<T | O3rElement> {
+    const elements = this.rootElement.sourceElement.element.locator(selector);
+    const element = elements.first();
+    const selectedElement: PlaywrightSourceElement = {element: element, page: this.rootElement.sourceElement.page};
+    return Promise.resolve(new (returnType || O3rElement)(selectedElement));
```

Comment:
in async function this is not required:

```suggestion
     return new (returnType || O3rElement)(selectedElement);
```

(same below)